### ISSUE TITLE
pb2mnn error

### ISSUE THREAD

**Issue by Wenboisstrong** [alibaba/MNN#882]
I'm a  MNN learner. I trained an image segmentation model and hope to use mnn as my mobile inference engine. When I think about it, the tensorflow pb model is converted to mnn. I get some errors. I hope anyone with experience can help me. 

~/CLionProjects/MNN/cmake-build-debug$ ./MNNConvert -f TF --modelFile model/frozen_model.pb --MNNModel mymodel.mnn --bizCode MNN

MNNConverter Version: 0.2.1.5git - MNN @ 2018

Start to Convert Other Model Format To MNN Model...
Start to Optimize the MNN Net...
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/Conv/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_2/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_2/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_3/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_3/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_4/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_4/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/Conv/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_1/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_2/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_3/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_4/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_5/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/expand/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/depthwise/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Don't support not const meanNode node 
Converte Tensorflow's Op mobilenet_v2/expanded_conv_6/project/BatchNorm/FusedBatchNorm , type = FusedBatchNorm, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_2/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_2/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_3/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_3/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_4/SpaceToBatchND , type = SpaceToBatchND, failed, may be some node is not const
Not Cost paddingShape
Converte Tensorflow's Op Conv_4/BatchToSpaceND , type = BatchToSpaceND, failed, may be some node is not const

terminate called after throwing an instance of 'Error'
  what():  [15:40:34] /home/kanbig/CLionProjects/MNN/tools/converter/source/common/writeFb.cpp:108: Check failed: (notSupportOps.size()) == (0) ==> These Op Not Support: Tensorflow::BatchToSpaceND | Tensorflow::FusedBatchNorm | Tensorflow::SpaceToBatchND | 
Aborted (core dumped)

**Comment by Wenboisstrong** [comment]
I learned from some search results on the Internet that I can try to set is_training = false to freeze the parameters of the BN layer. But I don’t quite understand why I did not freeze the parameters during the process of passing the ckpt 2 pb model

**Comment by czy2014hust** [comment]
Please pay attention to the place you put the flag(_set is_training = false_)

**Comment by Wenboisstrong** [comment]
@czy2014hust Thanks for your reply, I set is_training = false Tensorflow::FusedBatchNorm error is missing but still have the following error These Op Not Support: Tensorflow::BatchToSpaceND | Tensorflow::SpaceToBatchND | Do you have any suggestions to help me?

**Comment by Wenboisstrong** [comment]
I think I may have solved my problem, I refer to https://github.com/alibaba/MNN/issues/546
The answer fixed the size of my model input. This enabled me to successfully convert the model. Thank you @czy2014hust

**Comment by liuxx2016** [comment]
@Wenboisstrong i had the same problem. i would like to ask you where to set is_training = false? retraining or set during model change from .ckpt to .pb?

**Comment by liuxx2016** [comment]
@czy2014hust how and where to palce (set is_training = false)?
looking forward to your reply.

**Comment by djh123** [comment]
(set is_training = false)?  I use training one loop， in the end
